### ISSUE TITLE
Error when calling Web3App.createInstance()

### ISSUE THREAD

**Issue by kinggongzilla** [WalletConnect/Web3ModalFlutter#123]
**Describe the bug**
An error `LateError (LateInitializationError: Field '_web3App@594340753' has not been initialized.)` is thrown when calling Web3App.createInstance().

This error occurs with version web3modal_flutter version 3.1.3 

This error does **NOT** occur with web3modal_flutter version version: 3.0.21

The error is thrown in Line 73 in w3m_service.dart.

**To Reproduce**
1. Try to create new Web3App instance
Code:
```
   Web3App wcClient = await Web3App.createInstance(
    relayUrl: 'wss://relay.walletconnect.com',
    projectId: dotenv.env['WC_PROJECT_ID']!,
    metadata: const PairingMetadata(
      name: 'OwnerChip',
      description: 'OwnerChip - Connecting physical objects to the blockchain',
      url: 'https://www.ownerchip.com',
      icons: ['https://avatars.githubusercontent.com/u/116345848'],
    ),
  );
```

**Expected behavior**
New Web3App instance is created without errors.

**Smartphone (please complete the following information):**
 - Device: iPhone 13

**Comment by kinggongzilla** [comment]
Moreover I am encountering an issue when sending personal_sign request to Trust Wallet.

**Describe the bug**
After sending the personal_sign request, there is no popup in Trust Wallet to sign the request.
The issue does **NOT** exist with Metamask or 1inch wallet. 

**Steps to reproduce**
1. Connect Trust Wallet via Web3Modal
2. Send personal_sign request with:

```
String signature = await wc.request(
    topic: wcSession.topic,
    chainId: 'eip155:1',
    request: SessionRequestParams(
      method: 'personal_sign',
      params: [hexUtf8EncodedMessage, walletAddress.toString()],
    ),
  );
```

**Comment by kinggongzilla** [comment]
I'm using web3modal_flutter and we're using the modal for UI

**Comment by quetool** [comment]
But what are you using this code for?:

```
  Web3App wcClient = await Web3App wcClient = await Web3App.createInstance(
    relayUrl: 'wss://relay.walletconnect.com',
    projectId: dotenv.env['WC_PROJECT_ID']!,
    metadata: const PairingMetadata(
      name: 'OwnerChip',
      description: 'OwnerChip - Connecting physical objects to the blockchain',
      url: 'https://www.ownerchip.com',
      icons: ['https://avatars.githubusercontent.com/u/116345848'],
    ),
  );
```

This is how Web3Modal should be used 

https://docs.walletconnect.com/web3modal/flutter/options

**Comment by kinggongzilla** [comment]
Thanks for your quick reply. Sorry I left out a crucial part of the code.
I pass it to W3MService to create a new instance like this. 

```
final W3MService w3mService = W3MService(web3App: wcClient);
await w3mService.init();
```

**Comment by quetool** [comment]
Hello @kinggongzilla! You are right, I was able to reproduce. There's a bug in the code when using the instance through a web3app instance, however if you use it in the regular way you'll have no issues https://docs.walletconnect.com/web3modal/flutter/options#w3mservice-initialization

That would mean this
```dart
w3mService = W3MService(
  projectId: dotenv.env['WC_PROJECT_ID']!,
  metadata: const PairingMetadata(
    name: 'OwnerChip',
    description:
        'OwnerChip - Connecting physical objects to the blockchain',
    url: 'https://www.ownerchip.com',
    icons: ['https://avatars.githubusercontent.com/u/116345848'],
  ),
);
await w3mService!.init();
```

Instead of this
```dart
   Web3App wcClient = await Web3App.createInstance(
    relayUrl: 'wss://relay.walletconnect.com',
    projectId: dotenv.env['WC_PROJECT_ID']!,
    metadata: const PairingMetadata(
      name: 'OwnerChip',
      description: 'OwnerChip - Connecting physical objects to the blockchain',
      url: 'https://www.ownerchip.com',
      icons: ['https://avatars.githubusercontent.com/u/116345848'],
    ),
  );
```

**Comment by quetool** [comment]
GM @kinggongzilla! This bug have been fixed in latest 3.2.0 version! Thanks!
